### ISSUE TITLE
Low GPU Utilization (~40%) During Qwen2.5-VL-3B Training with mathv60k_message.jsonl Dataset

### ISSUE THREAD

**Issue by ymzUW** [TideDra/lmm-r1#76]
Hi team,

I'm training the Qwen2.5-VL-3B model on a single H100 GPU, using the mathv60k_message.jsonl dataset. However, I'm observing an average GPU utilization of around 40%.

There appears to be a significant idle time between the VLLM sampling stage and the actor forward stage. I’m wondering if this is expected behavior or indicative of a performance bottleneck or misconfiguration.
Here is the training script I'm using:
`ray job submit --address="http://127.0.0.1:8265" \
   --runtime-env-json='{"env_vars": {"VLLM_USE_V1": "1", "VLLM_ENABLE_V1_MULTIPROCESSING":"0"}}' \
   -- python -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 8 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 8 \
   --critic_num_nodes 1 \
   --critic_num_gpus_per_node 8 \
   --vllm_num_engines 8 \
   --vllm_tensor_parallel_size 1 \
   --remote_rm_url http://127.0.0.1:5000/get_reward \
   --colocate_all_models \
   --vllm_enable_sleep \
   --deepspeed_enable_sleep \
   --vllm_gpu_memory_utilization 0.5 \
   --vllm_sync_backend nccl \
   --enable_prefix_caching \
   --pretrain ${PRETRAIN_MODEL_PATH} \
   --save_path ${SAVE_PATH}/${MODEL_NAME} \
   --micro_train_batch_size 4 \
   --train_batch_size 256 \
   --micro_rollout_batch_size 8 \
   --rollout_batch_size 256 \
   --temperature 1.0 \
   --n_samples_per_prompt 16 \
   --max_epochs 1 \
   --num_episodes 2 \
   --prompt_max_len 6000 \
   --generate_max_len 2048 \
   --processor_kwargs '{"max_pixels": 3211264}' \
   --advantage_estimator reinforce_baseline \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 1e-6 \
   --init_kl_coef 0.001 \
   --prompt_data ${DATASET_PATH} \
   --input_key message \
   --label_key "answer" \
   --normalize_reward \
   --flash_attn \
   --lambd 1 \
   --gamma 1 \
   --gradient_checkpointing \
   --save_steps 20 \
   --ckpt_path ${SAVE_PATH}/${MODEL_NAME}/ckpt \
   --save_hf_ckpt \
   --load_checkpoint \
   --use_wandb ${WANDB_API_KEY} \
   --wandb_run_name ${MODEL_NAME} \
   --wandb_group "lmm-r1-training" \
   --use_tensorboard ${LOG_DIR} > >(tee -a "${CUR_LOG_DIR}/train.log") 2>&1 &`

The GPU utilization graph is this:
<img width="705" alt="Image" src="https://github.com/user-attachments/assets/1173a2c4-bc2d-4563-94e6-93f3f5ce3fee" />

**Comment by TideDra** [comment]
This is expected. The single controller needs to receive and process the samples and re-distribute them to the actors.

**Comment by ymzUW** [comment]
@TideDra Thanks for the reply! Do you have suggestions for reducing the GPU idle time and improving average gpu utilization?
